### PR TITLE
Exclusions: VisibleForTesting

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**: Removing exclusions stops new badness from creeping in.

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:
Will any of these break things for the large internal product?

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**: whenever

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java [hunk 1]
```diff
@@ -46,7 +46,6 @@
 
     private static final String PLACEHOLDER_TABLE_NAME = "{table}";
 
-    @VisibleForTesting
     public static final TableReference PLACEHOLDER_TABLE_REFERENCE =
```
Comment (Jolyon-S) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java:49]: is this technically only visible for testing? If so, can we just drop a comment here instead?

File: atlasdb-client/src/main/java/com/palantir/atlasdb/util/SlidingWindowMeanGauge.java [hunk 2]
```diff
@@ -39,6 +39,7 @@ public void update(long entry) {
         histogram.update(entry);
     }
 
+    @SuppressWarnings("VisibleForTestingPackagePrivate") // used in SweepMetricsAssert
```
Comment (Jolyon-S) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/SlidingWindowMeanGauge.java:42]: As above - `VisibleForTesting` does nothing other than let the developer know, so surely may as well just drop a comment instead?

File: atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ClientOptionsConstants.java [hunk 3]
```diff
@@ -16,13 +16,11 @@
 
 package com.palantir.atlasdb.http.v2;
 
-import com.google.common.annotations.VisibleForTesting;
 import com.palantir.conjure.java.api.config.service.HumanReadableDuration;
 
 public final class ClientOptionsConstants {
     static final HumanReadableDuration CONNECT_TIMEOUT = HumanReadableDuration.milliseconds(500);
 
-    @VisibleForTesting
```
Comment (Jolyon-S) [inline atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ClientOptionsConstants.java:25]: same

**Review by Jolyon-S — COMMENTED** [review]
I think we should not need to suppress the warnings, but just replace with comments where the annotation doesn't work

File: atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ClientOptionsConstants.java (hunk 3 above)
Comment (gsheasby) [reply atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ClientOptionsConstants.java:25]: this one is actually used in production code: `(Dialogue)ClientOptions`

Comment (Jolyon-S) [reply atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ClientOptionsConstants.java:25]: ha, fair enough

File: atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java (hunk 1 above)
Comment (Jolyon-S) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java:49]: no worries then - I guess the annotation was just bad here!

**Review by Jolyon-S — APPROVED** [review]

**Pull request merged**